ui/pages/Project/components/RnaSeqResultPage.jsx

```diff
@@ -0,0 +1,145 @@
+import { getRnaSeqDataLoading, getTissueOptionsByIndividualGuid } from '../selectors'
+
+const RnaSeqOutliers = React.lazy(() => import('./RnaSeqOutliers'))
+
+const TissueContainer = styled.div`
+  margin-bottom: 1rem;
+`
+
+const OUTLIER_VOLCANO_PLOT_CONFIGS = [
+  {
+    key: 'outliers',
+    getLocation: (({ geneId }) => geneId),
+    searchType: 'genes',
+    title: 'Expression Outliers',
+    formatData: data => data,
+  },
+  {
+    key: 'spliceOutliers',
+    getLocation: (({ chrom, start, end }) => `${chrom}:${Math.max(1, start - RNASEQ_JUNCTION_PADDING)}-${end + RNASEQ_JUNCTION_PADDING}`),
+    searchType: 'regions',
+    title: 'Splice Junction Outliers',
+    formatData: (data, tissueType) => data.filter(outlier => outlier.tissueType === tissueType),
+  },
+]
+
+class BaseRnaSeqResultPage extends React.PureComponent {
+
+  static propTypes = {
+    familyGuid: PropTypes.string,
+    rnaSeqData: PropTypes.object,
+    significantJunctionOutliers: PropTypes.arrayOf(PropTypes.object),
+    genesById: PropTypes.object,
+    tissueOptions: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  state = {
+    tissueType: null,
+  }
+
+  onTissueChange = (e, data) => {
+    this.setState({ tissueType: data.value })
+  }
+
+  render() {
+    const { familyGuid, rnaSeqData, significantJunctionOutliers, genesById, tissueOptions } = this.props
+    const { tissueType } = this.state
+    const showTissueType = tissueType || (tissueOptions?.length > 0 ? tissueOptions[0].value : null)
+    const tissueDisplay = TISSUE_DISPLAY[showTissueType]
+
+    const outlierPlotConfigs = OUTLIER_VOLCANO_PLOT_CONFIGS.map(({ formatData, ...config }) => ({
+      data: formatData(Object.values((rnaSeqData || {})[config.key] || {}).flat(), showTissueType),
+      ...config,
+    })).filter(({ data }) => data.length)
+
+    const outlierPlots = outlierPlotConfigs.map(({ key, data, ...config }) => (
+      <Grid.Column key={key} width={8}>
+        <RnaSeqOutliers
+          familyGuid={familyGuid}
+          rnaSeqData={data}
+          genesById={genesById}
+          {...config}
+        />
+      </Grid.Column>
+    ))
+
+    const tableData = significantJunctionOutliers.reduce(
+      (acc, outlier) => (outlier.tissueType === showTissueType ? [...acc, outlier] : acc), [],
+    )
+
+    return (
+      <div>
+        {showTissueType && (
+          <TissueContainer>
+            Tissue type: &nbsp;
+            {tissueOptions.length > 1 ? (
+              <Dropdown
+                text={tissueDisplay || 'Unknown Tissue'}
+                value={showTissueType}
+                options={tissueOptions}
+                onChange={this.onTissueChange}
+              />
+            ) : tissueDisplay}
+          </TissueContainer>
+        )}
+        <React.Suspense fallback={<Loader />}>
+          {(outlierPlots.length > 0) && (
+            <Grid>
+              <Grid.Row divided key={tissueType} columns={outlierPlots.length}>{outlierPlots}</Grid.Row>
```

Comment:
I added a `key` prop with a value of `tissueType`. The volcano plot won't be re-rendered after selecting a different tissue. React seems not smart enough to detect the change of `outlierPlots` after the `tissueType` changed. I don't know if there is any better way other than adding a `key` prop.

Reply:
Instead of this somewhat hacky fix, I think you should instead try doing a direct rendering of the plots here instead of assigning them to a variable and then putting that here. You are only using the variable to check the length, but since `outlierPlots` is a direct mapping of `outlierPlotConfigs` checking `outlierPlotConfigs.length` will get you the same thing. Then here you can do
```
 <Grid.Row divided columns={outlierPlotConfigs.length}>
  {outlierPlotConfigs.map(({ key, data, ...config }) => (
    <Grid.Column key={key} width={8}>
      ...
  )}
</Grid.Row>
```

Reply:
It still doesn't work. React doesn't know the `data` change when `tissueType` changes.

Reply:
yes, `componentDidMount` is only ever run once. Its designed to be used to handle high-computation things that should only be run once. It seems like you should probably change that to `componentDidUpdate`, with a check for `prevProps.data !== this.props.data` so it only re-renders when the data changes.

Reply:
I added `componentDidUpdate` to the `RnaSeqOutliersGraph` component. It looks like it is working now.